rocky/reports/views/base.py

```diff
@@ -202,8 +194,15 @@ def get_report_types(reports: set[type[BaseReportType]]) -> list[dict[str, str]]
             for report_type in reports
         ]
 
+    def report_type_selection_is_valid(self):
+        is_valid = True
+        if not self.selected_report_types:
+            is_valid = False
+            messages.error(self.request, _("Select at least one report type to proceed."))
+        return is_valid
```

Comment:
Also a getter with a side effect. See previous point about how to use such functions